### PR TITLE
Leaflet locations

### PR DESCRIPTION
**Pull request by uniphil**
Finally remove cartodb!

The regions view is still a bit rough, and the fly-out still needs to fade, but I don't think those block anything.

### CODE REVIEW

File: app/scripts/services/worldbankapi.coffee [hunk 1]
```diff
@@ -133,7 +133,28 @@ angular.module 'edudashAppSrv'
...
+        ].join ','
+        sql = "SELECT #{fields} FROM tz_#{educationLevel}_cleaned_dashboard"
         $http.get(wbApiRoot, {params: { q: sql, api_key: param1}})
 
+      getDistricts: (educationLevel) ->
+        table = if educationLevel is 'primary' then 'wbank.districts_primary' else 'wbank.secondary_districts'
+        sql = "
+          SELECT
+            district_n as name,
+            ST_AsGeoJSON(the_geom) as geojson
```
Comment (esoliani) [inline app/scripts/services/worldbankapi.coffee:155]: Good, nice way to get the geom as json!

File: app/views/dashboard.html [hunk 2]
```diff
@@ -82,21 +81,23 @@ <h3 translate="map.selected.average-pass-rate">Average Pass Rate</h3>
       <p class="number good">51</p>
     </div>
   </div>
-  <div map-flyout-school class="map-flyout" ng-if="mapView=='schools' && selectedSchool|exists"></div>
-  <!-- end map flyout -->
-
+  <div
+    class="map-flyout"
+    ng-show="hoveredSchool !== null"
+    ng-include="'views/schoolFlyout.html'">
```
Comment (esoliani) [inline app/views/dashboard.html:87]: This way is better that a directive without any custom code or scope

File: app/views/schoolFlyout.html [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  <h3>District</h3>
+  <p class="word">{{ hoveredSchool.district }}</p>
+</div>
+<div class="stat">
+  <h3>Ward</h3>
+  <p class="word">{{ hoveredSchool.ward }}</p>
+</div>
+<div class="stat">
+  <h3>Pass Rate</h3>
+  <p class="number {{ hoveredSchool.pass_2014 === null ?
```
Comment (esoliani) [inline app/views/schoolFlyout.html:19]: good one, I forget about the colors

File: test/spec/services/cartodb.coffee [hunk 4]
```diff
@@ -1,5 +0,0 @@
-'use strict'
-
-describe 'Service: cartodb', ->
```
Comment (esoliani) [inline test/spec/services/cartodb.coffee:3]: nice! remove cartodb

File: app/views/dashboard.html (hunk 2 above)
Comment (uniphil) [inline app/views/dashboard.html:87]: Just beware -- the path has to be quoted in the quotes, `="'blah'"`, because angular treats it as an expression

**Pull request merged**